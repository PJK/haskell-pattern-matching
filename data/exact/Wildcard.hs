module Wildcard where

data MyTrinary = A | B | C

data MyMaybe = Sum MyTrinary | Nope

extract :: MyMaybe -> MyMaybe-> MyTrinary
extract (Sum A) _ = A
extract (Sum B) _ = B
extract _ (Sum B) = B
extract _ (Sum C) = B
{-
    Uncovered:
    Sum C, Sum A
    Sum C, Nope
    Nope, Sum A
    Nope, Nope
-}
