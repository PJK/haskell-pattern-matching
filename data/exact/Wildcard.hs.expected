{
    "tag": "AnalysisSuccess",
    "contents": [
        [
            "extract",
            {
                "tag": "NonExhaustive",
                "contents": [
                    [
                        {
                            "tag": "ConstructorPattern",
                            "contents": [
                                "Sum",
                                [
                                    {
                                        "tag": "ConstructorPattern",
                                        "contents": [
                                            "C",
                                            []
                                        ]
                                    }
                                ]
                            ]
                        },
                        {
                            "tag": "ConstructorPattern",
                            "contents": [
                                "Sum",
                                [
                                    {
                                        "tag": "ConstructorPattern",
                                        "contents": [
                                            "A",
                                            []
                                        ]
                                    }
                                ]
                            ]
                        }
                    ],
                    [
                        {
                            "tag": "ConstructorPattern",
                            "contents": [
                                "Sum",
                                [
                                    {
                                        "tag": "ConstructorPattern",
                                        "contents": [
                                            "C",
                                            []
                                        ]
                                    }
                                ]
                            ]
                        },
                        {
                            "tag": "ConstructorPattern",
                            "contents": [
                                "Nope",
                                []
                            ]
                        }
                    ],
                    [
                        {
                            "tag": "ConstructorPattern",
                            "contents": [
                                "Nope",
                                []
                            ]
                        },
                        {
                            "tag": "ConstructorPattern",
                            "contents": [
                                "Sum",
                                [
                                    {
                                        "tag": "ConstructorPattern",
                                        "contents": [
                                            "A",
                                            []
                                        ]
                                    }
                                ]
                            ]
                        }
                    ],
                    [
                        {
                            "tag": "ConstructorPattern",
                            "contents": [
                                "Nope",
                                []
                            ]
                        },
                        {
                            "tag": "ConstructorPattern",
                            "contents": [
                                "Nope",
                                []
                            ]
                        }
                    ]
                ]
            }
        ]
    ]
}
